docs/internals/sync.md

```diff
@@ -0,0 +1,215 @@
+---
+sidebar_position: 3
+---
+
+# State Sync
+
+State sync synchronizes the state of a lagging/divergent replica with the healthy cluster.
+
+State sync is used when when a lagging replica's log no longer intersects with the cluster's current log —
+[WAL repair](./vsr.md#protocol-repair-wal) cannot catch the replica up.
+
+(VRR refers to state sync as "state transfer", but we already have [transfers](../reference/transfers.md) elsewhere.)
+
+In the context of state sync, "state" refers to:
+1. the grid (LSM data; acquired blocks only)
```

Comment:
Sorry, what does acquired blocks mean?

Reply:
A block (in the grid) is either _acquired_ or _free_. (This borrows terminology from the FreeSet. The FreeSet tracks which blocks are free/acquired, hence the name.)

Acquired blocks are part of the LSM tree. They are either:
1. a manifest log block (referenced by the superblock manifest)
2. a table index block (referenced by a manifest log block)
3. a table filter/data block (referenced by a table index block)

When a replica is first initialized, all of its blocks are free.
During compaction, the replica acquires blocks and writes to them.
And when a block isn't needed anymore, the block is freed, so that it can be reused later.

Reply:
Got it, thanks!